src/components/Posts/PostMenu.tsx

```diff
@@ -0,0 +1,130 @@
+import {
+  IconButton,
+  ListItemIcon,
+  ListItemText,
+  Menu,
+  MenuItem,
+} from "@mui/material";
+import React, { useState } from "react";
+import { useAppDispatch, useAppSelector } from "../../state/hooks";
+import { MoreVert, Edit, Delete, Link } from "@mui/icons-material";
+import axios from "axios";
+import { deletePost } from "../../state/slices/postsSlice";
+import { useNavigate } from "react-router-dom";
+
+type PostMenuProps = {
+  authorId: number;
+  postId: number;
+  isExpanded?: boolean;
+};
+
+const styles = {
+  listItemIcon: {
+    "&.MuiListItemIcon-root": { color: "black.main" },
+  },
+  listItemText: { fontWeight: "bold" },
+  menu: {
+    borderRadius: 4,
+  },
+};
+
+const PostMenu = ({ authorId, postId, isExpanded = false }: PostMenuProps) => {
+  const userId = useAppSelector((state) => state.user.userId);
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
+  const [menuOpen, setMenuOpen] = useState(false);
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+
+  const handleDelete = async () => {
+    try {
+      const result = await axios.delete(
+        `http://localhost:3001/api/posts/deletePost`,
+        {
+          data: {
+            postId: postId,
+            userId: userId,
+          },
+        }
+      );
+
+      isExpanded ? handleExpandedPostDelete() : handlePostDelete();
+      return result.data;
+    } catch (error) {
+      console.error("Failed to delete the post", error);
+    } finally {
+      handleMenuClose();
+    }
+  };
+
+  const handleMenuOpen = (event: React.MouseEvent<HTMLElement>) => {
```

Comment:
Similar here, my preference is to inline this in the `onClick` below

Reply:
I feel like using inline here reduces the readability of the code, and if I keep a handler for opening the menu, we could use it for any confirmation modal